homeassistant/components/romy/vacuum.py

```diff
@@ -0,0 +1,146 @@
+"""Support for Wi-Fi enabled ROMY vacuum cleaner robots.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/vacuum.romy/.
+"""
+
+
+from typing import Any
+
+from romy import RomyRobot
+
+from homeassistant.components.vacuum import StateVacuumEntity, VacuumEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, ICON, LOGGER
+from .coordinator import RomyVacuumCoordinator
+
+FAN_SPEED_NONE = "default"
+FAN_SPEED_NORMAL = "normal"
+FAN_SPEED_SILENT = "silent"
+FAN_SPEED_INTENSIVE = "intensive"
+FAN_SPEED_SUPER_SILENT = "super_silent"
+FAN_SPEED_HIGH = "high"
+FAN_SPEED_AUTO = "auto"
+
+FAN_SPEEDS: list[str] = [
+    FAN_SPEED_NONE,
+    FAN_SPEED_NORMAL,
+    FAN_SPEED_SILENT,
+    FAN_SPEED_INTENSIVE,
+    FAN_SPEED_SUPER_SILENT,
+    FAN_SPEED_HIGH,
+    FAN_SPEED_AUTO,
+]
+
+# Commonly supported features
+SUPPORT_ROMY_ROBOT = (
+    VacuumEntityFeature.BATTERY
+    | VacuumEntityFeature.PAUSE
+    | VacuumEntityFeature.RETURN_HOME
+    | VacuumEntityFeature.STATE
+    | VacuumEntityFeature.START
+    | VacuumEntityFeature.STOP
+    | VacuumEntityFeature.FAN_SPEED
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up ROMY vacuum cleaner."""
+
+    coordinator: RomyVacuumCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    romy: RomyRobot = coordinator.romy
+
+    device_info = {
+        "manufacturer": "ROMY",
+        "model": romy.model,
+        "sw_version": romy.firmware,
+        "identifiers": {"serial": romy.unique_id},
+    }
+
+    romy_vacuum_entity = RomyVacuumEntity(coordinator, romy, device_info)
+    entities = [romy_vacuum_entity]
+    async_add_entities(entities, True)
+
+
+class RomyVacuumEntity(CoordinatorEntity[RomyVacuumCoordinator], StateVacuumEntity):
+    """Representation of a ROMY vacuum cleaner robot."""
+
+    _attr_translation_key = DOMAIN
```

Comment:
```suggestion
    _attr_name = None
```

As the vacuum entity is the main entity for the device